tutorial/10_key_features/004_distributed.py

```diff
@@ -4,88 +4,306 @@
+            <TABLE BORDER="0" CELLBORDER="0" CELLALIGN="LEFT">
+                <TR><TD>Single node/<BR/>Multi-node?</TD></TR>
+            </TABLE>
+        >];
+
+        many_nodes  [label=<
+            <TABLE BORDER="0" CELLBORDER="0" CELLALIGN="LEFT">
+                <TR><TD>Do you need<BR/>a very large number of nodes?</TD></TR>
+            </TABLE>
+        >];
+
+        multithread_storages [
+            shape=box,
+            style=rounded,
+            href="#multi-thread-optimization",
+            label=<
+                <TABLE BORDER="0" CELLBORDER="0" CELLALIGN="LEFT">
+                    <TR><TD><U>InMemoryStorage</U></TD></TR>
+                    <TR><TD><U>JournalStorage</U></TD></TR>
+                </TABLE>
+            >
+        ];
+
+        singlenode_storages [
+            shape=box,
+            style=rounded,
+            href="#multi-process-optimization",
+            label=<
+                <TABLE BORDER="0" CELLBORDER="0" CELLALIGN="LEFT">
+                    <TR><TD><U>JournalStorage</U></TD></TR>
+                    <TR><TD><U>RDBStorage</U></TD></TR>
+                </TABLE>
+            >
+        ]
+
+        rdb_storage [
+            shape=box,
+            style=rounded,
+            href="#multi-node-optimization",
+            label=<
+                <TABLE BORDER="0" CELLBORDER="0" CELLALIGN="LEFT">
+                    <TR><TD><U>RDBStorage</U></TD></TR>
+                </TABLE>
+            >
+        ]
+
+        grpc_storage [
+            shape=box,
+            style=rounded,
+            href="#grpc-storage-proxy",
+            label=<
+                <TABLE BORDER="0" CELLBORDER="0" CELLALIGN="LEFT">
+                    <TR><TD><U>GrpcStorageProxy</U></TD></TR>
+                </TABLE>
+            >
+        ]
+
+        multithread -> multithread_storages [label="Multi-thread"];
+        multithread -> single_node [label="Multi-process"];
+        single_node -> singlenode_storages [label="Single node"];
+        single_node -> many_nodes [label="Multi-node"];
+        many_nodes -> rdb_storage [label="No"];
+        many_nodes -> grpc_storage [label="Yes"];
+    }
+
+.. _multi-thread-optimization:
+
+Multi-thread Optimization
+-------------------------
+
+.. Note::
+    **Recommended backends**:
+        - :class:`~optuna.storages.InMemoryStorage`
+        - :class:`~optuna.storages.JournalStorage`
+        - :class:`~optuna.storages.RDBStorage`
+
+
+You can run multiple trials in parallel just by setting the ``n_jobs``
+parameter in :func:`~optuna.create_study()`.
+
+Multi-thread optimization has traditionally been inefficient in Python due to the Global Interpreter Lock (GIL).
+However, starting from Python 3.14 (pending official release), the GIL is expected to be removed.
+This change will make multi-threading a good option, especially for parallel optimization.
 
 """
+
+import optuna
+from optuna.storages.journal import JournalStorage, JournalFileBackend
```

Comment:
`JournalStorage` can be imported from `optuna.stoarges`.
```suggestion
from optuna.storages import JournalStorage
from optuna.storages.journal import JournalFileBackend
```